Src/BootCamp.Chapter/Credentials.cs

```diff
@@ -2,13 +2,40 @@
 {
     internal struct Credentials
     {
-        public readonly string Username;
-        public readonly string Password;
+        private string _username;
 
-        public Credentials(string username, string password)
+        public string Username
         {
-            Username = username;
-            Password = password;
+            get
+            {
+                return _username;
+            }
+            set
+            {
+                if (value == null || value.Trim() == "")
```

Comment:
I like the validation inside a property. However, you should have used a built-in function for it:
```string.IsNullOrWhitespace(value)```